fastcrawler/test_utils.py

```diff
@@ -0,0 +1,62 @@
+from fastcrawler.engine import AioHttpEngine
+
+
+class TestServer:
+    def __init__(self):
+        self.responses = {
+            "GET": {"message": "GET request received"},
+            "POST": {"message": "POST request received"},
+            "PUT": {"message": "PUT request received"},
+            "DELETE": {"message": "DELETE request received"},
+        }
+
+    def get_response(self, method, data=None):
+        response = self.responses.get(method)
+        if data:
+            response["data"] = data
+        return response
+
+
+class TestClient(AioHttpEngine):
+    def __init__(self, server=TestServer(), headers=None, proxy=None):
+        self.server = server
+
+    def get(self, url):
+        response = self.server.get_response("GET")
+        return response
+
+    def post(self, url, data):
+        response = self.server.get_response("POST", data=data)
+        return response
+
+    def delete(self, url):
+        response = self.server.get_response("DELETE")
+        return response
+
+    def put(self, url, data):
+        response = self.server.get_response("PUT", data=data)
+        return response
+
+
+# prepare test
+
+server = TestServer()
+client = TestClient(server)
```

Comment:
That is good decoupling. but we don't have that in actual interface right? then respect the actual flow.
If you want to inject responses, then you should:

```python
mock_engine = MockEngine() # this just and only just mock the client
mock_client = MockClient(mapped_routes: dict, mock_engine) # this is high level, for more behaviour unsupported to mock the actual server and routing

router.get(...)
```